finetune/util/gpu_info.py

```diff
@@ -0,0 +1,36 @@
+import re
+import logging
+
+from tensorflow.python.client import device_lib
+
+LOGGER = logging.getLogger("finetune")
+
+def gpu_info(session_config=None):
+    def is_fp16_capable(device_desc):
+        match = re.search(r"compute capability: (\d+)\.(\d+)", device_desc)
+        if not match:
+            return False
+        return int(match.group(1)) >= 7
+    num_gpus = 0
+    fp16_capable = False
+    for local_device in device_lib.list_local_devices(session_config=session_config):
+        if local_device.device_type == "GPU":
+            num_gpus += 1
+            fp16 = is_fp16_capable(local_device.physical_device_desc)
+            if fp16 and not fp16_capable:
+                if num_gpus == 1:
+                    fp16_capable = True
+                else:
+                    LOGGER.warning(
+                        (
+                            "A GPU with float 16 support is available but it is not GPU:0,"
```

Comment:
What happens in the case of multiple gpus, some with float 16 support and others without?

Reply:
If it's too difficult to manage the various cases, we could just not support float16 unless all gpus that are visible have float16 enabled. Is that possible/easy to do?

Reply:
By default predict is put into the first gpu. So if this is not float 16, we don't use float 16, this warning is just to let people know that.

Reply:
:+1: